app/requirements/existential.py

```diff
@@ -0,0 +1,24 @@
+from plugins.stockpile.app.requirements.base_requirement import BaseRequirement
+
+
+class Requirement(BaseRequirement):
+
+    async def enforce(self, link, operation):
+        """
+        Given a link and the current operation, check that the operation's knowledge base contains
+        at least one fact (and edge) that complies with the requirement.
+        :param link
+        :param operation
+        :return: True if it complies, False if it doesn't
+        """
+        facts = await operation.all_facts()
+        filtered_facts = [fact for fact in facts if fact.trait == self.enforcements['source'] and
+                                                    link.paw in fact.collected_by]
+        if self.enforcements.get('edge', None):
+            filtered_facts = [fact for fact in filtered_facts if self._in_substring(self.enforcements['edge'], fact.relationships)]
+        if len(filtered_facts) > 0:
```

Comment:
maybe shorten if/else return to one liner:

"return bool(len(filtered_facts))"